app/api/server/v1/voip/rooms.ts

```diff
@@ -0,0 +1,182 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+import { Random } from 'meteor/random';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { API } from '../../api';
+import { VoipRoom, LivechatVisitors } from '../../../../models/server/raw';
+import { LivechatVoip } from '../../../../../server/sdk';
+import { IVoipRoom, OmnichannelSourceType } from '../../../../../definition/IRoom';
+/**
+ * @openapi
+ *  /voip/server/api/v1/voip/room <AMOL Verify during code review>
+ *    get:
+ *      description: Creates a new room if rid is not passed, else gets an existing room
+ * 		based on rid
+ * and token
+ *      security:
+ *      parameters:
+ *        - name: token
+ *          in: query
+ *          description: The visitor token
+ *          required: true
+ *          schema:
+ *            type: string
+ *          example: ByehQjC44FwMeiLbX
+ *        - name: rid
+ *          in: query
+ *          description: The room id
+ *          required: false
+ *          schema:
+ *            type: string
+ *          example: ByehQjC44FwMeiLbX
+ *        - name: agentId
+ *          in: query
+ *          description: Agent Id
+ *          required: false
+ *          schema:
+ *            type: string
+ *          example: ByehQjC44FwMeiLbX
+ *      responses:
+ *        200:
+ *          description: Room object and flag indicating whether a new room is created.
+ *          content:
+ *            application/json:
+ *              schema:
+ *                allOf:
+ *                  - $ref: '#/components/schemas/ApiSuccessV1'
+ *                  - type: object
+ *                    properties:
+ *                      room:
+ *                        type: object
+ *                        items:
+ *                          $ref: '#/components/schemas/IRoom'
+ *                      newRoom:
+ *                        type: boolean
+ *        default:
+ *          description: Unexpected error
+ *          content:
+ *            application/json:
+ *              schema:
+ *                $ref: '#/components/schemas/ApiFailureV1'
+ */
+API.v1.addRoute('voip/room', {
```

Comment:
Please configure the rate limiter for this endpoint to avoid abuse.

Reply:
Fixed. Thanks for this input. This is actually something that is affected by call volume. Good that instead of using default parameters, we are setting this explicitly.